dts/arm/nordic/nrf51822.dtsi

```diff
@@ -55,6 +55,7 @@
 			interrupts = <3 1>;
 			status = "disabled";
 			label = "I2C_0";
+			peripheral-number = <0>;
```

Comment:
Can you provide more details on the role of peripheral-number?

Reply:
Inside this driver we are using functions from nrfx, which take peripheral number as argument, this is a special case for that driver. Defines taken from dts start their numbering from 0 regardless of which peripheral is enabled, so if we only enable TWI instance 1 we will get following defines:
```
#define NORDIC_NRF5_I2C_0_LABEL "I2C_1"
#define NORDIC_NRF5_I2C_0_PERIPHERAL_NUMBER 1
```
This way we get constistent numbering and also we can use correct peripheral number as argument to nrfx functions.

Reply:
> Defines taken from dts start their numbering from 0 regardless of which peripheral is enabled

I think this may be confusing. Wouldn't it be better to use the node label instead of such numbering? Then, for the mentioned case we would get:
```
#define NORDIC_NRF5_I2C_I2C1_LABEL "I2C_1"
```
and there will be no need for the `peripheral-number` property.

Reply:
Firstly I wanted to keep define names unaffected by dts, and create them by using bindings only, so they all follow the same pattern.
Labels can be changed in dts, and sometimes it can be hard to predict them, for example in nrf52_pca20020 dts file:
```
sx1509b: sx1509b@3e {
	compatible = "semtech,sx1509b";
	reg = <0x3e>;
	label = "GPIO_P0";
};
```
Or some drivers use letters instead of numbers, e.g. GPIOA GPIOB
In such cases it can be hard to use define names with labels.
Also peripheral number won't be needed for sensor drivers or i2c/spi slave drivers, it really depends on how HAL is written for specific soc